devtools/x/src/lint/guppy.rs

```diff
@@ -507,3 +507,167 @@ impl PackageLinter for OnlyPublishToCratesIo {
         Ok(RunStatus::Executed)
     }
 }
+
+// Ensure that Move crates do not depend on Diem crates.
+#[derive(Debug)]
+pub struct MoveCratesDontDependOnDiemCrates;
+
+// List of existing move to diem dependencies.
+// The goal is eliminate all of them gradually.
+// In general, you should refrain from adding new entries to this list.
+const EXISTING_MOVE_TO_DIEM_DEPS: &[(&str, &str)] = &[
+    ("language-benchmarks", "language-e2e-tests"),
+    ("language-benchmarks", "diem-vm"),
+    ("language-benchmarks", "diem-types"),
+    ("language-benchmarks", "diem-state-view"),
+    ("language-benchmarks", "diem-proptest-helpers"),
+    ("bytecode-verifier", "diem-types"),
+    ("bytecode-verifier-tests", "diem-types"),
+    ("invalid-mutations", "diem-types"),
+    ("invalid-mutations", "diem-proptest-helpers"),
+    ("compiler", "diem-types"),
+    ("bytecode-source-map", "diem-types"),
+    ("ir-to-bytecode", "diem-types"),
+    ("ir-to-bytecode-syntax", "diem-types"),
+    ("compiled-stdlib", "diem-types"),
+    ("compiled-stdlib", "diem-proptest-helpers"),
+    ("compiled-stdlib", "diem-framework"),
+    ("compiled-stdlib", "diem-crypto"),
+    ("move-ir-types", "diem-types"),
+    ("move-lang", "fallible"),
+    ("move-lang", "diem-types"),
+    ("move-lang", "datatest-stable"),
+    ("move-lang-functional-tests", "functional-tests"),
+    ("move-lang-functional-tests", "diem-types"),
+    ("move-lang-functional-tests", "datatest-stable"),
+    ("move-lang-test-utils", "datatest-stable"),
+    ("move-model", "diem-types"),
+    ("move-model", "datatest-stable"),
+    ("move-prover", "diem-types"),
+    ("move-prover", "diem-temppath"),
+    ("move-prover", "datatest-stable"),
+    ("abigen", "diem-types"),
+    ("abigen", "diem-temppath"),
```

Comment:
Whats the plan when we're using common deps like this? migrate to something on crates.io?

Reply:
Yeah we'll likely migrate to the crates.io versions.

Reply:
Is diem-temppath the only example? As I recall that's a really minimal wrapper function for temppath, so maybe we should just get rid of it? :)

Reply:
Also `fallible`, `diem-fallible`, `num-variants` and `datatest-stable`? `diem-proptest-helpers` might also be tricky...

Reply:
aside from datatest i think you should be able to depend directly on existing published crates or just handle the logic yourself in the move crates

Reply:
Yeah these dependencies shouldn't be a big problem to us. We have many choices.